Dev/csv export 8122

mcantelon opened the pull request:
This PR adds bulk CSV export of descriptions, either RAD or ISAD, and does a little bit of cleanup/refactoring of the bulk export class. A parent CSV export class exists upon which other CSV export child classes can be created.

mcantelon committed:
Minor refactor of export base class. refs #8122

Changed the bulk export base class so the format option can be added
by child classes (the option's text description will vary for
different types of bulk exports: XML vs CSV for example).

Made other minor changes to make it applicable to non-XML formats.

mcantelon committed:
Added bulk CSV export. refs #8122

Added a task, and related classes, for exporting descriptions as CSV.
CSV data can either include RAD or ISAD-related sets of columns.

mcantelon committed:
Fixed minor CSV date import issue. refs #8122

Fixed an issue that caused CSV to import improper dates.

MikeFE commented on lib/flatfile/QubitFlatfileExport.class.php:
```diff
@@ -0,0 +1,421 @@
+      $filename = exportBulkBaseTask::generateSortableFilename($this->fileIndex, 'csv', $this->standard);
+      $filePath = $this->path .'/'. $filename;
+    }
+    else {
```
minor style issue, braces on their own lines always

MikeFE commented on lib/flatfile/csvInformationObjectExport.class.php:
```diff
@@ -0,0 +1,390 @@
+
+    foreach($properties as $property)
+    {
+      array_push($alternativeIdentifierLabels, $property->name);
```
http://php.net/manual/en/function.array-push.php seems to indicate in the note if you're just pushing a single value to an array, $ary[] = 123 is slightly better than array_push(). But I don't think it's a big deal.

MikeFE commented on lib/job/arBaseJob.class.php:
```diff
@@ -135,7 +135,7 @@ private function formatLogMsg($message)
    * Clear various Qubit classes' caches.
    */
-  private function clearCache()
+  static function clearCache()
```
I think this method makes more sense at this point put into lib/helper/QubitHelper.php as something named clear_all_orm_cache() or similar. It seems kind of weird to be in arBaseJob if it's being used way elsewhere in the code base.

mcantelon replied:
Agreed. I'll look into making it a helper.

mcantelon committed:
Add helper for clearing object caches. refs #8122

Added a helper function that will clear in-memory caches of instantiated
objects maintained by various classes.
